sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/CSVOptions.scala

```diff
@@ -192,6 +192,20 @@ class CSVOptions(
    */
   val emptyValueInWrite = emptyValue.getOrElse("\"\"")
 
+  /**
+   * A string between two consecutive JSON records.
+   */
+  val lineSeparator: Option[String] = parameters.get("lineSep").map { sep =>
+    require(sep.nonEmpty, "'lineSep' cannot be an empty string.")
+    require(sep.length == 1, "'lineSep' can contain only 1 character.")
```

Comment:
I currently have a project where we are importing windows newlines CRLF from CSV files. 

I backported these changes but ran into an issue with this check, because to properly parse Windows CSV files I must be able to set "\r\n" for lineSep in the settings.

It appears the reason this require was added is no longer needed as the code for asReaderSettings/asWriterSettings never calls that function anymore.

I was able to remove this assert and now able to import the windows newline CSV files into dataframes properly now.

Another issue I had before this was the very last column would always get a "\r" at the end of the column name, so something like "TEXT" would become "TEXT\r", and therefore we would be unable to query the TEXT column anymore. Setting lineSep to "\r\n" solved this issue as well.

Reply:
>  I must be able to set "\r\n" for lineSep in the settings.

I don't need to set `\r\n` to `lineSep` to split an input by lines because Hadoop Line Reader can detect `\r\n` itself. In which mode do you parse the CSV files - per-line `multiLine = false` or multiline?

Reply:
I am setting multiLine = "true".

The problem I am having with this is the column name of the last column in the CSV header gets a \r added to the end of it.

So if I have

name,age,text\r\nfred,30,"likes\r\npie,cookies,milk"\njill,30,"likes\ncake,cookies,milk"\r\n

I was getting schema with 

StringType("NAME")
IntegerType("AGE")
StringType("TEXT\r")

Could it be the mixed use of \r\n and \n so it only wants to use \n for newlines?

Another issue is the configuration for lineSep is controlled upstream from a different configuration provided by users who have no knowledge of spark, but know how they formatted their CSV files, and without some re-architecture, it is not possible to detect that this setting is set to \r\n and then set it to None for the CSVOptions.

lineSeparator.foreach(format.setLineSeparator) already handles 1 to 2 characters so I figured this is a safe thing to support for lineSep configuration no?

Reply:
For multiline true, we have fixed auto-multiline detect feature in CSV (see https://github.com/apache/spark/pull/22503) That will do the job.

Reply:
That is taken care of in this by the following line that I backported no?

`settings.setLineSeparatorDetectionEnabled(lineSeparatorInRead.isEmpty && multiLine)`

I am still having the issue that univocity keeps a \r in the column name with multiline set to True and lineSeparatorInRead is unset.

The only way I seem to be able to get spark to not put a \r in the column name is to specifiy the lineSep option with two characters explicitly to `\r\n`. Then I get a normal set of column names and everything else parses correctly.

Reply:
Would you be able to file a JIRA after testing out against the master branch if the issue is persistent?